### DISCUSSION TITLE
Event named "do".

### DISCUSSION THREAD

**Discussion by cfjedimaster** [alpinejs/alpine#4237] [7. Bugs (archived)]
I added this to a button, `@click="do(cat)"` and got an error: 

```
SyntaxError: Unexpected token 'do'
```

Now, I admit, do is not a great name for a function, I had planned to give it a better name later, but it seems like only "do" causes this issue. I tested with aa, bb, and multiple other ones, and doIt is fine too. It seems that just "do" is an issue. Any reason why?

**Comment by SimoTod** [comment]
do is a reserved word in javascript, you can't call a function like that.
https://www.w3schools.com/js/js_reserved.asp
That's not an Alpine bug.

**Reply by cfjedimaster** [reply]
Ah duh, I haven't used a do while loopnin ages. Thank you.
